src/compiler/types.ts

```diff
@@ -1007,6 +1013,10 @@ namespace ts {
         expression: LeftHandSideExpression;
         typeArguments?: NodeArray<TypeNode>;
     }
+    export interface SupportedExpressionWithTypeArguments extends ExpressionWithTypeArguments {

```

Comment:
Looks like `isSupportedExpressionWithTypeArguments` is only ever called immediately before calling `node.expression`, so maybe I should get rid of the `SupportedExpressionWithTypeArguments` type?